2_process.R

```diff
@@ -425,14 +425,43 @@ p2_targets_list <- list(
                     #covered by physiographic regions
                     #RUN7100 seems like it is by HUC02 instead of reach.
                     #RFACT is perfectly correlated with RF7100
-                    drop_columns = c("PHYSIO_AREA", "RUN7100", "RFACT")),
+                    drop_columns = c("PHYSIO_AREA", "RUN7100", "RFACT"))
+  ),
   
-  # Match PRMS stream segments to synoptic site ids and return subset of sites within 
+  # Match PRMS stream segments to synoptic and flow site ids and return subset of sites within 
   # the distance specified by search_radius (in meters)
   tar_target(
     p2_syn_sites_w_segs,
     get_site_flowlines(p1_reaches_sf, p1_syn_sites, sites_crs = 4269, max_matches = 1, 
                        search_radius = bird_dist_cutoff_m, retain_sites = NULL)
+  ),
+  tar_target(
+    p2_flow_sites_w_segs,
+    get_site_flowlines(p1_reaches_sf, p1_flow_sites, sites_crs = 4269, max_matches = 1, 
+                       search_radius = 100, retain_sites = NULL)
+  ),
+  
+  #Get candidate sampling segments with road density >= 6 or NPDES >= 5
+  tar_target(
+    p2_candidate_syn_sites,
+    {PRMS_segs <- c(filter(p2_nhdv2_attr_refined, TOT_TOTAL_ROAD_DENS >= 6) %>% 
+                             pull(PRMS_segid),
+                           filter(p2_nhdv2_attr_refined, CAT_NPDES_MAJ_sum >= 5) %>% 
+                             pull(PRMS_segid)) %>% 
+                           unique()
+    #Candidate sites
+    syn_sites <- filter(p2_flow_sites_w_segs, subsegid %in% PRMS_segs) %>%
+      #make sure they're not already sampled
+      filter(!(subsegid %in% p2_syn_sites_w_segs$subsegid)) %>%
+      #add attributes
+      left_join(p2_nhdv2_attr_refined %>% 
+                  select(PRMS_segid, TOT_TOTAL_ROAD_DENS, CAT_NPDES_MAJ_sum), 
+                by = c("subsegid" = "PRMS_segid"))
+    
+    write_csv(syn_sites, file = '2022_DRBCandidateSynopticSites.csv')
+    '2022_DRBCandidateSynopticSites.csv'
```

Comment:
It looks like this file would save to the root project directory rather than one of the phase 'out' files. What do you think about replacing these lines with the following:

```suggestion
    syn_sites_fileout <- '2_process/out/2022_DRBCandidateSynopticSites.csv'
    write_csv(syn_sites, file = syn_sites_fileout)
    syn_sites_fileout
```